x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/state_management/simulation_state_machine/simulation_state_machine.ts

```diff
@@ -173,20 +173,14 @@ export const simulationMachine = setup({
         actions: [{ type: 'storeSamples', params: ({ event }) => event }],
       },
     ],
-    'step.change': {
-      target: '.debouncingChanges',
-      reenter: true,
-      description: 'Re-enter debouncing state and reinitialize the delayed processing.',
-      actions: [{ type: 'storeSteps', params: ({ event }) => event }],
-    },
-    'step.delete': [
+    'simulation.updateSteps': [
       {
         guard: {
           type: 'hasSteps',
           params: ({ event }) => ({ steps: event.steps }),
         },
-        target: '.assertingRequirements',
         actions: [{ type: 'storeSteps', params: ({ event }) => event }],
+        target: '.debouncingChanges',
```

Comment:
Is this required under what scenario? The difference in targeting directly the `.assertingRequirements` state is that for some immediate interaction (like click on a button, etc.) we don't need debouncing, which causes a slight delay.
Has this changes somwhere?